### PR TITLE
feat: Implement SQLite-Python destination plugin.

### PR DESCRIPTION
**Pull request by marianogappa**
This is meant as the simplest destination plugin so that we have an example destination plugin in Python.

Missing:
- release-please, etc, not yet done.
- SDK changes need to be merged first.
- I haven't been able to do the round-trip test yet; it requires SDK changes that are tricky; but I'm pretty sure read/write work well now.

Note second run relies on state to do virtually nothing:
```
Starting sync for: tenable (cloudquery/tenable@v2.2.4) -> [sqlite (grpc@localhost:7777)]
Sync completed successfully. Resources: 214303, Errors: 0, Warnings: 0, Time: 5m15s
✓ Code/test-tenable-sqlite-sync  $ cli sync .                                                  ⏱ 16:46:49
Loading spec(s) from .
Starting sync for: tenable (cloudquery/tenable@v2.2.4) -> [sqlite (grpc@localhost:7777)]
Sync completed successfully. Resources: 19, Errors: 0, Warnings: 0, Time: 5s
✓ Code/test-tenable-sqlite-sync  $
```

```
$ cli sync .                                               ⏱ 16:35:35
Loading spec(s) from .
Starting sync for: xkcd (cloudquery/xkcd@v1.0.6) -> [sqlite-python (grpc@localhost:7777)]
Sync completed successfully. Resources: 2973, Errors: 0, Warnings: 0, Time: 20s
✓ Code/test-xkcd-sqlitepython-sync  $
```

State stored and read properly
Screenshot 2024-08-20 at 17 03 16

Dates & Extension types managed properly
Screenshot 2024-08-20 at 17 05 05

### CODE REVIEW

File: plugins/destination/sqlite-python/Makefile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+fmt:
+	black .
+
+fmt-check:
+	black --check .
+
+build-docker:
+	docker build -t sqlite-python:latest .
+
+gen-docs: build-docker
```
Comment (erezrokah) [inline plugins/destination/sqlite-python/Makefile:15]: `gen-docs` is a source plugin thing, no need for it here

File: plugins/destination/sqlite-python/plugin/client/client.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+from plugin.sqlite.client import SQLClient, SQLiteColumn
+from cloudquery.sdk.schema import Table
+import pyarrow as pa
+# from plugin.client.type_conversions import arrow_type_to_sqlite_str
+from cloudquery.sdk import message
+
+METADATA_TABLE_NAME = "cq:table_name"
+CQ_SYNC_TIME_COLUMN = "cq_sync_time"
+CQ_SOURCE_NAME_COLUMN = "cq_source_name"
```
Comment (erezrokah) [inline plugins/destination/sqlite-python/plugin/client/client.py:17]: We should expose these from the SDK

File: plugins/destination/sqlite-python/Makefile (hunk 1 above)
Comment (marianogappa) [reply plugins/destination/sqlite-python/Makefile]: Removed

File: plugins/destination/sqlite-python/plugin/client/client.py (hunk 2 above)
Comment (marianogappa) [reply plugins/destination/sqlite-python/plugin/client/client.py]: Moved to the SDK.

**Comment by marianogappa** [comment]
Tested in docker:
```
$ cli sync .                                               ⏱ 16:23:58
Loading spec(s) from .
Starting sync for: xkcd (cloudquery/xkcd@v1.0.6) -> [sqlite-python@docker (sqlitepython:latest@)]
Sync completed successfully. Resources: 2973, Errors: 0, Warnings: 0, Time: 28s
```

File: .github/workflows/dest_sqlite_python.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    branches:
+      - main
+    paths:
+      - "plugins/destination/sqlite-python/**"
+      - ".github/workflows/dest_sqlite_python.yml"
+
+jobs:
+  plugins-destination-sqlite-python:
+    timeout-minutes: 30
+    name: "plugins/source/sqlite-python"
```
Comment (erezrokah) [inline .github/workflows/dest_sqlite_python.yml:22]:
```suggestion
    name: "plugins/destination/sqlite-python"
```

File: .github/workflows/dest_sqlite_python.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      - ".github/workflows/dest_sqlite_python.yml"
+
+jobs:
+  plugins-destination-sqlite-python:
+    timeout-minutes: 30
+    name: "plugins/destination/sqlite-python"
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./plugins/source/sqlite-python
```
Comment (erezrokah) [inline .github/workflows/dest_sqlite_python.yml:26]:
```suggestion
        working-directory: ./plugins/destination/sqlite-python
```

File: release-please-config.json [hunk 5]
```diff
@@ -61,6 +61,9 @@
     "plugins/destination/sqlite": {
       "component": "plugins-destination-sqlite"
     },
+    "plugins/destination/sqlite-python": {
+      "component": "plugins-destination-sqlite-python"
+    },
```
Comment (erezrokah) [inline release-please-config.json:66]:
```suggestion
```

Let's not release it for now, we can keep the plugin as a reference for a developer guide we'll add to https://docs.cloudquery.io/docs/developers/creating-new-plugin

**Pull request merged**